rest/doc_api.go

```diff
@@ -266,48 +268,56 @@ func (h *handler) handlePutDoc() error {
 	}()
 
 	docid := h.PathVar("docid")
-	body, err := h.readDocument()
-	if err != nil {
-		return err
-	}
-	if body == nil {
-		return base.ErrEmptyDocument
-	}
 
 	var newRev string
 	var doc *db.Document
 	var ok bool
+	var err error
+	var body db.Body
 
 	roundTrip := h.getBoolQuery("roundtrip")
 
-	if h.getQuery("new_edits") != "false" {
-		// Regular PUT:
-		if oldRev := h.getQuery("rev"); oldRev != "" {
-			body[db.BodyRev] = oldRev
-		} else if ifMatch := h.rq.Header.Get("If-Match"); ifMatch != "" {
-			body[db.BodyRev] = ifMatch
-		}
-		newRev, doc, err = h.db.Put(docid, body)
+	if replicator2, _ := h.getOptBoolQuery("replicator2", false); replicator2 {
+		newRev, doc, err = h.handlePutDocReplicator2(docid)
```

Comment:
Now that we're switching to handlePutDocReplicator2 early, I expected we wouldn't need any other changes in handlePutDoc.  It looks like there still a lot going on below - can this be reverted?

Reply:
There's not really that many changes there at all. Its just git showing the diff a bit funny. For the most part its just the old stuff moved into an `else`. There's still a little similarity between the two at the end of `handlePutDoc` so didn't want to branch entirely as there would be duplication which is why I went with `else` rather than early return.

Reply:
If you clone lines 322-328 into handlePutDocReplicator2, you can just call 'return h.handlePutDocReplicator2(docid) here, and avoid touching the rest of PutDoc altogether.  I think that's definitely preferable, as it avoids adding code complexity and regression potential.